lib/host.py

```diff
@@ -534,43 +536,53 @@ def management_pif(self):
+        disks = self.ssh(['lsblk', '--noheadings', '--nodeps',
+                          '-I', '8,259', # limit to: sd, blkext
+                          '--output', 'NAME']).splitlines()
         disks.sort()
         return disks
 
-    def raw_disk_is_available(self, disk: str) -> bool:
+    def disk_is_available(self, disk: DiskDevName) -> bool:
         """
-        Check if a raw disk (without any identifiable filesystem or partition label) is available.
+        Check if a disk is unmounted and appears available for use.
+
+        It may or may not contain identifiable filesystem or partition label.
+        If there are no mountpoints, it is assumed that the disk is not in use.
 
-        It suggests the disk is "raw" and likely unformatted thus available.
+        Warn: This function may misclassify LVM_member disks (e.g. in XOSTOR, RAID, ZFS) as "available".
+        Such disks may not have mountpoints but still be in use.
         """
-        return self.ssh_with_result(['blkid', '/dev/' + disk]).returncode == 2
+        return len(self.ssh(['lsblk', '--noheadings', '--nodeps', '-o', 'MOUNTPOINT', '/dev/' + disk]).strip()) == 0
 
-    def disk_is_available(self, disk: str) -> bool:
+    def disk_is_4k(self, disk: DiskDevName) -> bool:
         """
-        Check if a disk is unmounted and appears available for use.
+        Check if a disk has 4k blocks.
 
         It may or may not contain identifiable filesystem or partition label.
         If there are no mountpoints, it is assumed that the disk is not in use.
 
         Warn: This function may misclassify LVM_member disks (e.g. in XOSTOR, RAID, ZFS) as "available".
         Such disks may not have mountpoints but still be in use.
         """
-        return len(self.ssh(['lsblk', '-n', '-o', 'MOUNTPOINT', '/dev/' + disk]).strip()) == 0
+        return self.ssh(['lsblk', '--noheadings', '--nodeps',
+                         '-o', 'LOG-SEC',
+                         '/dev/' + disk]).strip() == '4096'
 
-    def available_disks(self, blocksize=512):
+    def available_disks(self, blocksize: int = 512) -> list[DiskDevName]:
         """
         Return a list of available disks for formatting, creating SRs or such.
 
         Returns a list of disk names (eg.: ['sdb', 'sdc']) that don't have any mountpoint in
         the output of lsblk (including their children such as partitions or md RAID devices)
         """
         avail_disks = []
-        blk_output = self.ssh(['lsblk', '-nd', '-I', '8,259', '--output', 'NAME,LOG-SEC']).splitlines()
-        for line in blk_output:
-            line = line.split()
+        blk_output = self.ssh(['lsblk', '--noheadings', '--nodeps',
+                               '-I', '8,259', # limit to: sd, blkext
+                               '--output', 'NAME,LOG-SEC']).splitlines()
+        for line_str in blk_output:
+            line = line_str.split()
             disk = line[0]
             sec_size = line[1]
```

Comment:
```suggestion
        for line in blk_output:
            disk, sec_size = line.split()
```

Reply:
Separate optimization, but yes. Not strictly equivalent if `lsblk` ever breaks the assumption, but should be safe.

Reply:
applied